yetanotherco/aligned_layer
# feat: pay to batcher with keystore

# Description

- This PR allows the client to pay to the batcher.

# To Test

- Import/Create Keystore file, you can follow README_SEND_PROOFS.md instructions for this
- Make sure you have the latest version of the aligned contracts deployed.
- Run `make install_aligned_compiling`
- Run `make anvil_start`.
- Fund your wallet:

```
cast send <address> --value 1ether --rpc-url http://localhost:8545/ --private-key 0x2a871d0798f97d79848a013d4936a73bf4cc922c825d33c1cf7073dff6d409c6
```

- Send to the batcher:

```
aligned deposit-to-batcher --keystore_path <keystore_path> --amount 0.1ether
```

- Check your balance in the batcher:

```
cast call 0x7969c5eD335650692Bc04293B07F5BF2e7A673C0 "UserBalances(address)(uint256)" <address>
```

It should be **100000000000000000**.

Depends on #460.
